packages/xo-web/src/xo-app/backup-ng/index.js

```diff
@@ -66,15 +67,26 @@ const SchedulePreviewBody = ({ item: job, userData: { schedulesByJob } }) => (
           />
         </td>
         <td>
-          <ActionButton
-            btnStyle='primary'
-            data-id={job.id}
-            data-name={job.name}
-            data-schedule={schedule.id}
-            handler={_runBackupNgJob}
-            icon='run-schedule'
-            size='small'
-          />
+          {job.runId !== undefined ? (
+            <ActionButton
+              btnStyle='danger'
+              handler={cancelJob}
+              handlerParam={job}
+              icon='cancel'
+              size='small'
+              tooltip={_('formCancel')}
+            />
+          ) : (
+            <ActionButton
+              btnStyle='primary'
+              data-id={job.id}
+              data-name={job.name}
+              data-schedule={schedule.id}
+              handler={_runBackupNgJob}
+              icon='run-schedule'
+              size='small'
+            />
+          )}
```

Comment:
Because it's the same component used at the same place in the tree, React think it's the same button, and, as a consequence, share the state with both of them.

The issue is that when the job is run via the button, the button goes into a pending state, preventing the job from being canceled.

A work-around can be to use the `key` prop.